src/Modules/CurrentMapModule/Services/CurrentMapService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using EvoSC.Common.Database.Repository.Maps;
+using EvoSC.Common.Interfaces;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.CurrentMapModule.Interfaces;
+using Microsoft.Extensions.Logging;
+using GbxRemoteNet.Events;
+using ISO3166;
+
+namespace EvoSC.Modules.Official.CurrentMapModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Transient)]
+public class CurrentMapService : ICurrentMapService
+{
+    private readonly ILogger<CurrentMapService> _logger;
+    private readonly IManialinkManager _manialinkManager;
+    private readonly MapRepository _mapRepository;
+    private readonly IServerClient _client;
+
+    public CurrentMapService(IManialinkManager manialinkManager,
+        ILogger<CurrentMapService> logger,
+        MapRepository mapRepository, IServerClient client)
+    {
+        _logger = logger;
+        _manialinkManager = manialinkManager;
+        _mapRepository = mapRepository;
+        _client = client;
+    }
+
+    private static string GetCountry(string Zone)
+    {
+        var zones = Zone.Split("|");
+        if (zones.Length > 2)
+        {
+            return zones[2];
+        }
+
+        return "Other";
+    }
+
+    private async Task ShowManialinkAsync(string mapUId)
+    {
+        var dbMap = await _mapRepository.GetMapByUidAsync(mapUId);
+        List<Country> countries = Country.List.ToList();
+        var country = countries.Find(country => country.Name == GetCountry(dbMap?.Author?.Zone ?? ""));
+
+        await _manialinkManager.SendPersistentManialinkAsync("CurrentMapModule.CurrentMapWidget",
+            new { map = dbMap, country = country?.ThreeLetterCode ?? "WOR" });
+        _logger.LogDebug("Showing widget");
```

Comment:
Would add some more information to this debug statement. E.g. what widget is actually being shown
```suggestion
        _logger.LogDebug("Showing Current Map widget");
```